traits/trait_types.py

```diff
@@ -3675,10 +3675,27 @@ def __init__(self, **metadata):
     def validate(self, object, name, value):
         """ Raises an error, since no values can be assigned to the trait.
         """
-        raise TraitError(
-            "The '%s' trait of %s instance is a read-only "
-            "UUID." % (name, class_of(object))
-        )
+        if not self.can_init:
```

Comment:
Where is `can_init` added as an attribute for this class?

Reply:
It's not at the moment. It's assumed to be passed in via the metadata. Where should the attribute be declared?

Reply:
It's just that looking at the code for `TraitType` doesn't indicate that metadata will be turned into attributes on the trait instance. The tests passed, so the magic must be somewhere, right?

Reply:
Dug a bit more... Looks like `TraitType.as_ctrait` is the magic.

Reply:
Actually, it's `TraitType.__getattr__()`